### PR TITLE
Fix #87: Don't create duplicate Enums for fields with choices

### LINKED ISSUE #87: Bug with Django ChoiceField
Hi, hi 
I got one bug using `models.CharField` with choices. 
Graphene is creating two differents objects as `Enum`. 
Raising one error in `graphene.types.typemap`

```
File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 64, in reducer
    return self.graphene_reducer(map, type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 82, in graphene_reducer
    return self.construct_objecttype(map, type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 153, in construct_objecttype
    map[type._meta.name]._fields = self.construct_fields_for_type(map, type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 214, in construct_fields_for_type
    map = self.reducer(map, field.type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 64, in reducer
    return self.graphene_reducer(map, type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 82, in graphene_reducer
    return self.construct_objecttype(map, type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 153, in construct_objecttype
    map[type._meta.name]._fields = self.construct_fields_for_type(map, type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 214, in construct_fields_for_type
    map = self.reducer(map, field.type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 64, in reducer
    return self.graphene_reducer(map, type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 69, in graphene_reducer
    return self.reducer(map, type.of_type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 64, in reducer
    return self.graphene_reducer(map, type)
  File "/usr/local/lib/python2.7/dist-packages/graphene-1.1.3-py2.7.egg/graphene/types/typemap.py", line 79, in graphene_reducer
    assert _type.graphene_type == type
AssertionError
********************************************************************************
_type.graphene_type  UserMobile1Os 139869850885568
type UserMobile1Os 139869850674512
********************************************************************************
```

My model

```
MOBILE_PLATFORMS_CHOICES = (
    ('android', 'Android'),
    ('ios', 'iOS'),
)


class User(AbstractUser):
    full_name = models.CharField(max_length=64, blank=True)
    mobile_1_os = models.CharField(max_length=10, default='android',
                                   choices=MOBILE_PLATFORMS_CHOICES,
                                   db_index=True)

```

My query

```
class Query(graphene.AbstractType):
    users = DjangoConnectionField(
        UserNode
    )
```

My `UserNode` 

```
class UserNode(DjangoObjectType):
    class Meta(object):
        interfaces = (relay.Node,)
        model = User
```

And I have one file for `Schemas`

```
class AllQuery(Query, ObjectType):

   viewer = graphene.Field(lambda: AllQuery)

    @classmethod
    def resolve_viewer(cls, *args, **kwargs):
        return AllQuery()

schema_graphene = graphene.Schema(query=AllQuery)
```

### PR DESCRIPTION
**Pull request by aaxelb**
If `convert_django_field_with_choices` is called multiple times on a field with choices, a new Enum will be created each time, which breaks graphene's TypeMap. (#87)

I fixed this by storing converted Django fields in the registry, so converting the same field multiple times will return exactly the same converted object.

### CODE REVIEW

File: graphene_django/registry.py [hunk 1]
```diff
@@ -2,7 +2,7 @@ class Registry(object):
 
     def __init__(self):
         self._registry = {}
-        self._registry_models = {}
```
Comment (aaxelb) [inline graphene_django/registry.py]: `_registry_models` is not referenced anywhere, so it seemed safe to remove.

**Comment by AshwinRamesh** [comment]
Is the fix by @aaxelb going to be merged in soon? Currently can't upgrade to 1.3 :(

**Comment by paultiplady** [comment]
This build failure is just on the `pypy` tests, and looks like a flake in the test runner: 

`/home/travis/.travis/job_stages: line 62: py.test: command not found`

And pypy tests pass for me against this branch when I run manually.

Worth a core dev retrying the build pipeline, or @aaxelb force-pushing a new commit to this branch to trigger another run?

More generally, is there any interest from the core team in taking this fix, or some feedback on what would need to change to be accepted? This issue is blocking me from using graphene in my company, so happy to assist if possible.

**Comment by paultiplady** [comment]
Nice one, all 💚  now.

**Comment by jm2242** [comment]
very excited to see this fix go live!

**Pull request merged**